combat-trainer.lic

```diff
@@ -295,73 +295,74 @@ class LootProcess
     end
   end
 
-  def dispose_body(game_state)
-    return unless @loot_bodies
-    if DRRoom.dead_npcs.empty?
-      @last_ritual = nil
-      return
-    end
-    return if Time.now - @loot_timer < @loot_delay
+  def check_rituals(game_state)
+    return unless DRStats.necromancer?
 
-    game_state.mob_died = true
-    healing = game_state.casting_consume || game_state.prepare_consume
-    if @necro && @last_ritual.nil?
+    if @last_ritual.nil?
       if @heal
         game_state.wounds = check_health
         echo "Severity to Wounds: #{game_state.wounds}" if $debug_mode_ct
         unless game_state.wounds.empty?
-          game_state.prepare_consume = healing = check_thanatology(DRRoom.dead_npcs.first, 'preserve', game_state) && check_thanatology(DRRoom.dead_npcs.first, 'consume', game_state)
+          do_ritual(DRRoom.dead_npcs.first, 'preserve', game_state)
+          do_ritual(DRRoom.dead_npcs.first, 'consume', game_state)
         end
       end
-      if @ritual_type == 'harvest'
-        check_thanatology(DRRoom.dead_npcs.first, 'preserve', game_state)
-      end
-      check_thanatology(DRRoom.dead_npcs.first, @ritual_type, game_state) unless healing
-      if ['preserve'].include?(@ritual_type) && !healing
-        arrange_mob(DRRoom.dead_npcs.first, game_state)
-        check_skinning(DRRoom.dead_npcs.first, game_state)
-      end
-    elsif @skin && !%w(consume harvest dissect).include?(@last_ritual)
-      arrange_mob(DRRoom.dead_npcs.first, game_state)
-      check_skinning(DRRoom.dead_npcs.first, game_state)
-    end
-    unless healing
-      while 'and get ready to search it' == bput('loot', 'You search', 'I could not find what you were referring to', 'and get ready to search it')
-        pause
-        waitrt?
+
+      if @ritual_type == 'harvest' && ! game_state.prepare_consume
```

Comment:
Are these compound if statements clear? Should nest all other possible rituals inside a single `unless game_state.prepare_consume` instead?